docs/adr/0024-upgrade-templating-toolset-of-auth-server-emails.md

```diff
@@ -0,0 +1,177 @@
+- MJML helps significantly with responsiveness in emails, reducing time spent developing templates and making future email redesigns easier
+
+### Email Previewing
+
+Chosen option: "Option B - Use Storybook", because:
+
+- It is easy to setup and provides us the flexibility to preview all of the different email states together.
+- We have a Storybook deployment already in place and hooking it up for `fxa-auth-server` will ensure consistency.
+- Easy to test and perform QA without needing to touch the codebase.
+- Support for CSS (whether it's plain old CSS, TailwindCSS or something else)
+
+Note: Along with using storybook to view the various states of the templates, we are also planning on continued support of [Maildev](https://www.npmjs.com/package/maildev) since it has applications beyond just previewing emails.
+
+### Other
+
+**Plaintext files**: We'll use [html-to-text](https://github.com/html-to-text/node-html-to-text) to automatically generate plaintext versions of templates rendered to HTML. There may be a scenario in which the automatically-generated plaintext version does not look exactly how we'd like, in which case we can look into exporting both an MJML _and_ plaintext version of the email from the template file.
+
+**Localization**: We will upgrade the localization tool from GetText to [Fluent](https://github.com/projectfluent/fluent.js) since it's preferred by the l10n team and other FxA front-ends are using it. With our chosen templating option we can make use of Fluent's [`@fluent/dom`](https://github.com/projectfluent/fluent.js/tree/master/fluent-dom) package.
+
+[**Integration & QA**: During development templates will be marked as being part of a release group. This could be in the form of a mapped list of template names or some variable associated with the template file. Each group will have a corresponding environment variable flag. When this release group’s flag is enabled any templates that fall under it will be used when generating an email template in that environment; until a template's corresponding release group is enabled it is not used and the old/current template will continue to be served. Release groups will be initially enabled in staging until QA has had an opportunity to thoroughly test and evaluate each template in the group, after which they can be incrementally enabled in production.] **Vijay to edit**
+
+### Positive Consequences
+
+- Using MJML abstracts HTML email peculiarities away and handles responsiveness for us
+- Allows us to move away from inline CSS during development, and improves style reusability and consistency in template files
+- Using EJS allows us to write templates using JavaScript, removing the complexities of custom syntaxes like JSX and Mustache
+
+### Negative Consequences
+
+- The Storybook setup will be much more complex than a React and Tailwind solution
```

Comment:
Lol, I wrote this sentence, but I think it might be more clear to use the word _with_: The Storybook setup will be much more complex than with a React and Tailwind solution